src/qcardia_data/setup/reformat/utils.py

```diff
@@ -0,0 +1,303 @@
+    start_idx_1, stop_idx_1 = borderless_idxs_1[0], borderless_idxs_1[-1] + 1
+
+    cine_borderless = cine.slicer[start_idx_0:stop_idx_0, start_idx_1:stop_idx_1, ...]
+    gt_borderless = gt.slicer[start_idx_0:stop_idx_0, start_idx_1:stop_idx_1, ...]
+
+    orientations = nib.orientations.io_orientation(cine_borderless.affine)
+
+    # force third dimension to stay there
+    if orientations[2, 0] == 0.0:
+        orientations[:2, 0] -= 1
+        orientations[2, 0] = 2.0
+    elif orientations[2, 0] == 1.0:
+        orientations[orientations[:, 0] == 2.0, 0] = 1.0
+        orientations[2, 0] = 2.0
+
+    orientations[:2, 0] = orientations[:2, 0][::-1]  # swap first and second axes
+    if process_orientations_func is not None:
+        orientations = process_orientations_func(orientations, gt_borderless)
+
+    cine_processed = set_nifti_zooms(cine_borderless.as_reoriented(orientations))
+    gt_processed = set_nifti_zooms(gt_borderless.as_reoriented(orientations))
+
+    return cine_processed, gt_processed
+
+
+def get_frame_index(cine: np.ndarray, frame: np.ndarray) -> int:
+    """Get the index of a 3D time frame in a 4D cine scan.
+
+    Args:
+        cine (np.ndarray): 4D cine scan with the timeframe dimension as the last
+        frame (np.ndarray): 3D time frame to find in the cine scan
+
+    Returns:
+        int: timeframe index of the frame in the cine scan
+    """
+    for frame_nr in range(cine.shape[-1]):
+        if np.abs(np.sum(cine[..., frame_nr] - frame)) <= 0.0:
+            return frame_nr
+
+
+def get_ed_es_idxs(cine_gt: np.ndarray) -> tuple[int, int]:
+    """Get the annotated ED and ES timeframe indexes from a 4D cine ground truth mask.
+
+    Args:
+        cine_gt (np.ndarray): 4D ground truth mask with labeled ED and ES timeframes,
+            with the timeframe dimension as the last
+
+    Returns:
+        tuple[int, int]: ED and ES timeframe indexes respectively
+    """
+    idx = np.nonzero(np.sum(cine_gt, axis=(0, 1, 2)))[0]  # two annotated frame indexes
+
+    # get volumes of the two annotated frames
+    vols = np.array(
+        [
+            np.sum(cine_gt[..., idx[0]] > 0),
+            np.sum(cine_gt[..., idx[1]] > 0),
+        ]
+    )
+
+    ed_est = idx[np.argmax(vols)]  # frame with the largest volume is ED
+    es_est = idx[np.argmin(vols)]  # frame with the smallest volume is ES
+    return ed_est, es_est
+
+
+def reorder_df_columns(df: pd.DataFrame) -> pd.DataFrame:
+    """reorder the columns of a dataframe to have the SubjectID and Pathology columns
+    first.
+
+    Args:
+        df (pd.DataFrame): dataframe with SubjectID and Pathology columns, and other
+            columns
+
+    Returns:
+         pd.DataFrame: dataframe with SubjectID and Pathology columns first
+    """
+    reordered_cols = ["SubjectID", "Pathology"]
+    reordered_cols += [col for col in df.columns.to_list() if col not in reordered_cols]
+    return df[reordered_cols]
+
+
+def load_dicom_data(folder):
```

Comment:
weird that this is here already but happy to keep it

Reply:
Yeah I figured as all DICOM stuff will be for reference anyway as our DICOM data isn't public, this function could already be included in this version.